### PR TITLE
Fixing branch

### PR DESCRIPTION
**Pull request by polikarpov13**

### CODE REVIEW

File: src/main/java/book/store/onlinebookstore/dto/order/PlaceOrderRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package book.store.onlinebookstore.dto.order;
+
+import jakarta.validation.constraints.NotNull;
+
+public record PlaceOrderRequestDto(
+        @NotNull
+        String shippingAddress) {
+}
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/onlinebookstore/dto/order/PlaceOrderRequestDto.java]:
```suggestion
public record PlaceOrderRequestDto(@NotNull String shippingAddress) {
}

```

File: src/main/java/book/store/onlinebookstore/dto/order/UpdateOrderStatusRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package book.store.onlinebookstore.dto.order;
+
+import jakarta.validation.constraints.NotNull;
+
+public record UpdateOrderStatusRequestDto(
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/onlinebookstore/dto/order/UpdateOrderStatusRequestDto.java]: same, with 1 param better fo it in 1 row

File: src/main/java/book/store/onlinebookstore/enums/Status.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        }
+    }
+
+    public static Status valueOf(int value) {
+        return statusMap.get(value);
+    }
+
+    public int getValue() {
+        return value;
+    }
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/onlinebookstore/enums/Status.java]: it is overengineering, leave only enum values, and make it a nested class in the Order model

File: src/main/java/book/store/onlinebookstore/dto/order/UpdateOrderStatusRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package book.store.onlinebookstore.dto.order;
+
+import jakarta.validation.constraints.NotNull;
+
+public record UpdateOrderStatusRequestDto(
+        @NotNull
+        int statusValue) {
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/onlinebookstore/dto/order/UpdateOrderStatusRequestDto.java]:
```suggestion
        String status) {

```

File: src/main/java/book/store/onlinebookstore/service/order/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            orderItem.setQuantity(item.getQuantity());
+            orderItem.setPrice(item.getBook().getPrice());
+
+            orderItemRepository.save(orderItem);
+            orderItems.add(orderItem);
+        }
+
+        order.setOrderItems(orderItems);
+        return orderMapper.toDto(order);
+    }
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/onlinebookstore/service/order/OrderServiceImpl.java]: Is it possible to reduce complexity of this method? maybe extract some logical parts into private methods

File: src/main/java/book/store/onlinebookstore/service/order/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    public OrderDto placeOrder(PlaceOrderRequestDto placeOrderRequestDto) {
```
Comment (Vasul-Pankiv) [inline src/main/java/book/store/onlinebookstore/service/order/OrderServiceImpl.java:38]: Make this method transactional. Think about what else methods should be transactional

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/onlinebookstore/mapper/OrderItemMapper.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import book.store.onlinebookstore.config.MapperConfig;
+import book.store.onlinebookstore.dto.orderitem.OrderItemDto;
+import book.store.onlinebookstore.model.OrderItem;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class, componentModel = "spring")
+public interface OrderItemMapper {
+
```
Comment (kozhukhovsky) [inline src/main/java/book/store/onlinebookstore/mapper/OrderItemMapper.java:12]:
```suggestion

```

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**